### PR TITLE
codegen+runtime: Make C++ output easier to parse visually

### PR DESCRIPTION
**Pull request by cg-jl**
- codegen: Hint when parentheses are needed in codegen_expression
- codegen: Avoid generating extra parentheses with error handlers
- codegen: Use `this->` on accessors when appropiate

Codegen added a lot of extra opening parentheses that made expressions inside `while`/`if` hard to parse visually due to a large number of opening parentheses. Now codegen does not emit parentheses on most cases that it doesn't need to, it's only missing cases that are based on binary operator precedence.
These commits removed ~2000 parentheses and ~100kB worth of selfhost C++ code.

- runtime: Make integral comparisons use C++ operators

Due to the inlined comparison functions that used `.compare` with a cast, any Jakt expression that had a `<` (e.g the condition for `while bytes_written < writes_read`) would emit two nested IIFEs because it had two infallible conversions to and from `Ordering` plus a comparison to the `Ordering` enum. Now the integral comparisons are explicitly overloaded to use `<` and the like so the compiler doesn't generate the IIFEs that make the conditions in `while` and `if` unreadable.
This commit alone manages to suppress another ~100kB worth of selfhost C++ code.

The new compile times are within the noise threshold of the old ones.
The new generated code does not affect the C++ parser, but the resulting C++ code is smaller, easier to read and more intuitive to search through. This in turn makes checking whether a modification in codegen.jakt was successful or not faster, since the code is more greppable.

### CODE REVIEW

**Comment by alimpfard** [comment]
Tests are failing because of some AK changes, [this](https://github.com/SerenityOS/jakt/pull/1595/commits/a40d2a27670eff6b5a0171e203afe25f7e55183b#diff-2ccdf1410db0d3a7d32bd68071c6060e12f221adfed757de4bf2e022565bafebR156), [this](https://github.com/SerenityOS/jakt/pull/1595/commits/a40d2a27670eff6b5a0171e203afe25f7e55183b#diff-16fda0fe34bd7d97af9a1ba1024ecb3d1eaffc229fa314881e1924e97e62461fR1) and [this](https://github.com/SerenityOS/jakt/pull/1595/commits/a40d2a27670eff6b5a0171e203afe25f7e55183b#diff-9713ae70fac2072b1e57293f3f8bdfa3a03c0facce95a99395e9040f2c280488R1) should fix it.
Feel free to yoink those into your PR if you want.

**Comment by cg-jl** [comment]
oh, thanks! I forgot that I didn't have an up to date serenity repo 😅

**Comment by alimpfard** [comment]
Very nice, looks great!
will merge after CI passes.
